packages/framework/ai-collab/src/test/explicit-strategy/promptGeneration.spec.ts

```diff
@@ -0,0 +1,411 @@
+		view.initialize(initialAppState);
+
+		idGenerator.assignIds(view.root);
+		const actualPromptWithEmptyEditLogAndPlan = getEditingSystemPrompt(
+			userAsk,
+			idGenerator,
+			view.root,
+			[],
+			systemRoleContext,
+			plan,
+		).split("\n");
+		const expectedPromptWithEmptyEditLogAndPlan = getExpectedEditingSystemPrompt({
+			plan,
+			userAsk,
+			editLog: [],
+			treeNode: view.root,
+		});
+
+		assert.deepStrictEqual(
+			actualPromptWithEmptyEditLogAndPlan,
+			expectedPromptWithEmptyEditLogAndPlan,
+		);
+	});
+
+	it("Editing System Prompt with plan and populated edit log has no regression", () => {
+		const tree = factory.create(
+			new MockFluidDataStoreRuntime({ idCompressor: createIdCompressor() }),
+			"tree",
+		);
+		const view = tree.viewWith(new TreeViewConfiguration({ schema: TestTodoAppSchema }));
+		view.initialize(initialAppState);
+
+		idGenerator.assignIds(view.root);
+
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const todo1Id = idGenerator.getId(view.root.todos[0]!)!;
+
+		const editLog: EditLog = [
+			// We expect an error for this edit because the field is 'completed' not 'complete'
+			{
+				edit: {
+					type: "modify",
+					explanation:
+						"Change the completed field to false for the todo at index 0 in the list of todos",
+					target: { target: todo1Id },
+					field: "complete",
+					modification: false,
+				},
+			},
+			{
+				edit: {
+					type: "modify",
+					explanation:
+						"Change the completed field to false for the todo at index 0 in the list of todos",
+					target: { target: todo1Id },
+					field: "completed",
+					modification: false,
+				},
+			},
+		];
+		const simpleSchema = getSimpleSchema(Tree.schema(view.root));
+		for (const editLogEntry of editLog) {
+			try {
+				applyAgentEdit(editLogEntry.edit, idGenerator, simpleSchema.definitions);
+			} catch (error) {
+				assert(error instanceof Error);
+				editLogEntry.error = error.message;
+			}
+		}
+
+		const actualPromptWithPlanAndEditLog = getEditingSystemPrompt(
+			userAsk,
+			idGenerator,
+			view.root,
+			editLog,
+			systemRoleContext,
+			plan,
+		).split("\n");
+
+		const expectedPromptWithPlanAndEditLog = getExpectedEditingSystemPrompt({
+			plan,
+			userAsk,
+			editLog,
+			treeNode: view.root,
+		});
+		assert.deepStrictEqual(actualPromptWithPlanAndEditLog, expectedPromptWithPlanAndEditLog);
+
+		// Another test for an edit log with a single edit
+		const actualPromptWithPlanAndSingleEditEditLog = getEditingSystemPrompt(
+			userAsk,
+			idGenerator,
+			view.root,
+			editLog,
+			systemRoleContext,
+			plan,
+		).split("\n");
+
+		const expectedPromptWithPlanAndSingleEditEditLog = getExpectedEditingSystemPrompt({
+			plan,
+			userAsk,
+			editLog,
+			treeNode: view.root,
+		});
+		assert.deepStrictEqual(
+			actualPromptWithPlanAndSingleEditEditLog,
+			expectedPromptWithPlanAndSingleEditEditLog,
+		);
```

Comment:
FYI I removed this in a follow up commit, this is not testing anything different